pkg/cloud/services/eks/roles.go

```diff
@@ -204,6 +204,14 @@ func (s *NodegroupService) reconcileNodegroupIAMRole() error {
 	}
 
 	policies := NodegroupRolePolicies()
+	if s.scope.ManagedMachinePool.Spec.RoleAdditionalPolicies != nil {
```

Comment:
Could we do a **len** check here instead of a nil check?

Reply:
ack